lnbits/core/views/api.py

```diff
@@ -359,19 +363,19 @@ async def api_payments_sse(
     request: Request, wallet: WalletTypeInfo = Depends(get_key_type)
 ):
     return EventSourceResponse(
-        subscribe(request, wallet), ping=20, media_type="text/event-stream"
+        subscribe(request, wallet.wallet), ping=20, media_type="text/event-stream"
     )
 
 
 @core_app.get("/api/v1/payments/{payment_hash}")
 async def api_payment(payment_hash, X_Api_Key: Optional[str] = Header(None)):
-    wallet = None
-    try:
-        if X_Api_Key.extra:
-            print("No key")
-    except:
-        wallet = await get_wallet_for_key(X_Api_Key)
+    # We use X_Api_Key here because we want this call to work with and without keys
+    wallet = await get_wallet_for_key(X_Api_Key) if X_Api_Key is not None else None
```

Comment:
Does exactly what you've described before (@arcbtc) but cleaner. Is there a reason for the `X_Api_Key.extra` check?